Upstream

michaeldewildt opened the pull request:
Gday Ben,

Thanks heaps for writing this fantastic API! I have been using in my WordPress plugin and it has been working great. It turnes out it can work with PHP 5.2.16 or higher so I have back ported my version and simplified it a bit for better reuse.

Anyways I thought I would contribute some of my tweaks back upstream. 

**1. Ignore DS_Store on OSX and convert tabs to four spaces where spacing was off.**

They are both a tad annoying :-)

**2. Ensure the path has a trailing slash when uploading chunked uploads.**

Without it the file will be uploaded with a concatenated folder and file name.

**3. Raise an exception if a cURL error occurs**

The API raises Dropbox errors but not curl errors. I found this out the hard way on a flight the other day. 

I hope this helps!

Cheers,
Mikey

Commit:
Ignore DS_Store on OSX

Commit:
Convert tabs to four spaces

Commit:
Ensure the path has a trailing slash

Commit:
Raise an exception if a cURL error occures

BenExile commented:
Hi Michael,

Thank you for your contribution. Gah, tabs were due to my new IDE install - thanks.

I'm happy to merge all but the `usleep()` added to the chunked upload method. This is because I am working on integrating a callback function which will be fired after each chunk has been uploaded. I have had no problems so far with alignment issues due to Dropbox not processing the chunk in time for the next call, and so if you wish to use `usleep()`, you will be able to in the callback.

Please update your pull request to reflect the above and I'll merge in straight away :)

michaeldewildt commented:
Thanks mate.

I removed the usleep. To be fair im not 100% positive is necessary with your recent chunked upload updates, I added it as a discussion point. :-)

I have some PHPUnit tests too that I will push up soon.

Cheers,
Mikey

BenExile commented:
No problem. Thanks for your work. Unit tests are what this library is lacking at the moment and I've just not had the time - anything you've got would be much appreciated!
